runtime/compilers/rillv1/parse_model.go

```diff
@@ -106,6 +112,44 @@ func (p *Parser) parseModel(node *Node) error {
 		inputProps["sql"] = sql
 	}
 
+	// special handling to mark model as updated when local file changes
+	if inputConnector == "local_file" {
+		c, ok := inputProps["invalidate_on_change"].(bool)
+		invalidateModelOnUpdate := ok && c
+		path, ok := inputProps["path"].(string)
+		if ok && invalidateModelOnUpdate {
+			entries, err := p.Repo.ListRecursive(ctx, path, true)
+			if err == nil && len(entries) > 0 {
+				var localPaths []string
+				root, err := p.Repo.Root(ctx)
+				if err != nil {
+					return err
+				}
+
+				// Update parser's localDataToResourcepath map to track which resources depend on the local file
+				for _, entry := range entries {
+					localPaths = append(localPaths, filepath.Join(root, entry.Path))
+					resources, ok := p.localDataToResourcepath[entry.Path]
+					if !ok {
+						resources = make(map[string]any)
```

Comment:
The nesting is getting very deep here, consider pulling the `inputConnector == "local_file"` handling into a separate function to support early return: https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting